### PR TITLE
Add new permission to allow editing content owner

### LINKED ISSUE #13212: Add settings to CommonPart to determine which roles can see the CommonPart
### Is your feature request related to a problem? Please describe.
Currently, `SiteOwner` permission is required to be able to see/edit the `CommonPart` info like owner. I want to allow a moderator to change the owner of a specific content type. Currently, there is no way to do that without granting moderators `SiteOwner` which isn't acceptable.

### Describe the solution you'd like
Add a setting to the `CommonPart` that the user can select which roles can manage the `CommonPart` in addition to `SiteOwner`. We have to auto grant `SiteOwner` permission to keep it backward compatible.

### PR DESCRIPTION
**Pull request by MikeAlhayek**
Fix #13212

### Updated
Now we have `EditContentOwner` and `EditContentOwner_{}` permission to control who can edit the owner of content item when `CommonPart` is attached.

### CODE REVIEW

**Comment by sebastienros** [comment]
Try to use permissions, even if you need one per type.

**Comment by hishamco** [comment]
How is different from choosing an `Administrator` role?

**Comment by MikeAlhayek** [comment]
@sebastienros I changed the PR to use permissions instead of settings as per your request.

File: src/OrchardCore.Modules/OrchardCore.Contents/Drivers/OwnerEditorDriver.cs [hunk 1]
```diff
@@ -37,7 +37,8 @@ public override async Task<IDisplayResult> EditAsync(CommonPart part, BuildPartE
         {
             var currentUser = _httpContextAccessor.HttpContext?.User;
 
-            if (!await _authorizationService.AuthorizeAsync(currentUser, StandardPermissions.SiteOwner, part))
+            if (!await _authorizationService.AuthorizeAsync(currentUser, StandardPermissions.SiteOwner)
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Contents/Drivers/OwnerEditorDriver.cs:40]: Try without the SiteOwner it should work. Or/and add it to the Administrator role.

**Review by sebastienros — APPROVED** [review]

**Pull request merged**